### PR TITLE
Feature/refactor william

### PR DESCRIPTION
**Pull request by wjeam**

### CODE REVIEW

File: backend/src/main/java/com/team4/backend/model/ProfileImage.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+    @Id
+    private String id;
+    private Binary image;
+    private String fileName;
+    private String uploaderEmail;
+    @Builder.Default
+    private LocalDateTime creationDate = LocalDateTime.now();
+
+    @Builder(builderMethodName = "profileImageBuilder")
```
Comment (0xD34DC0DE) [inline backend/src/main/java/com/team4/backend/model/ProfileImage.java:28]: No need for custom builder name if not extending another class

File: backend/src/main/java/com/team4/backend/service/FileMetaDataService.java [hunk 2]
```diff
@@ -28,10 +33,16 @@
 
     private final FileAssetService fileAssetService;
 
-    public FileMetaDataService(FileMetaDataRepository fileMetaDataRepository, StudentService studentService, FileAssetService fileAssetService) {
+    private final UserService userService;
+
+    private final NotificationService notificationService;
+
+    public FileMetaDataService(FileMetaDataRepository fileMetaDataRepository, StudentService studentService, FileAssetService fileAssetService, UserService userService, NotificationService notificationService) {
```
Comment (0xD34DC0DE) [inline backend/src/main/java/com/team4/backend/service/FileMetaDataService.java:40]: put arguments on their own line

File: backend/src/main/java/com/team4/backend/service/InternshipOfferService.java [hunk 3]
```diff
@@ -30,14 +32,20 @@
...
 
+    private final NotificationService notificationService;
+
+    private final UserService userService;
+
     public InternshipOfferService(InternshipOfferRepository internshipOfferRepository,
                                   MonitorService monitorService,
                                   StudentService studentService,
-                                  SemesterService semesterService) {
+                                  SemesterService semesterService, NotificationService notificationService, UserService userService) {
```
Comment (0xD34DC0DE) [inline backend/src/main/java/com/team4/backend/service/InternshipOfferService.java:42]: Arguments on new lines

File: frontend/src/components/ProfileImageUpload.js [hunk 4]
```diff
@@ -0,0 +1,369 @@
...
+              </Grid>
+            </Paper>
+          </Grid>
+        </Grid>
+      </div>
+    </motion.div>
+  );
+};
+
+export default ProfileImageUpload;
```
Comment (Maxirus98) [inline frontend/src/components/ProfileImageUpload.js:369]: Le Composant est huge, serait-il possible de le décomposer?

Comment (Maxirus98) [reply frontend/src/components/ProfileImageUpload.js:369]: 369 lignes

File: frontend/src/components/OfferForm.js [hunk 5]
```diff
@@ -32,6 +32,7 @@ const OfferForm = () => {
   const [dialog, dialogDispatch] = useContext(DialogContext);
 
   const handleFormChange = (event) => {
+    console.log(offer);
```
Comment (Maxirus98) [inline frontend/src/components/OfferForm.js:35]: console log

File: backend/src/main/java/com/team4/backend/service/ProfileImageService.java [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+                .map(User::getEmail)
+                .collect(Collectors.toSet())
+                .flatMapMany(userEmails ->
+                        profileImageRepository
+                                .findByUploaderEmails(userEmails)
+                                .collectList()
+                                .flatMapMany(profileImages ->
+                                        Flux.zip(
+                                                        Flux.fromIterable(profileImages),
+                                                        Flux.fromIterable(profileImages)
```
Comment (0xD34DC0DE) [inline backend/src/main/java/com/team4/backend/service/ProfileImageService.java:61]: wtf

File: backend/src/test/java/com/team4/backend/controller/MonitorControllerTest.java [hunk 7]
```diff
@@ -82,4 +80,35 @@ public void shouldNotCreateMonitor() {
...
+
+        //ACT
+        webTestClient
+                .get()
+                .uri("/monitor/getProfile")
+                .exchange()
+                //ASSERT
+                .expectStatus()
+                .isNotFound()
+                .expectBody(MonitorProfileDto.class);
```
Comment (0xD34DC0DE) [inline backend/src/test/java/com/team4/backend/controller/MonitorControllerTest.java:111]: returns a body of type MonitorProfileDto even if not found ?

File: backend/src/test/java/com/team4/backend/controller/SupervisorControllerTest.java [hunk 8]
```diff
@@ -206,4 +207,35 @@ void shouldNotGetSupervisor() {
...
+
+        //ACT
+        webTestClient
+                .get()
+                .uri("/supervisor/getProfile")
+                .exchange()
+                //ASSERT
+                .expectStatus()
+                .isNotFound()
+                .expectBody(SupervisorProfileDto.class);
```
Comment (0xD34DC0DE) [inline backend/src/test/java/com/team4/backend/controller/SupervisorControllerTest.java:238]: same thing

File: backend/src/test/java/com/team4/backend/mapping/InternshipManagerMapperTest.java [hunk 9]
```diff
@@ -47,4 +51,22 @@ void mapDtoToEntity() {
...
+    @Test
+    void mapEntityToProfileDto() {
+        //ARRANGE
+        InternshipManager entity = InternshipManagerMockData.getInternshipManager();
+
+        //ACT
+        InternshipManagerProfileDto dto = InternshipManagerMapper.toProfileDto(entity);
+
+        //ASSERT
+
```
Comment (0xD34DC0DE) [inline backend/src/test/java/com/team4/backend/mapping/InternshipManagerMapperTest.java:63]: extra line

File: backend/src/test/java/com/team4/backend/mapping/MonitorMapperTest.java [hunk 10]
```diff
@@ -49,4 +50,23 @@ void mapDtoToEntity() {
...
+    @Test
+    void mapEntityToProfileDto() {
+        //ARANGE
+        Monitor entity = MonitorMockData.getMockMonitor();
+
+        //ACT
+        MonitorProfileDto dto = MonitorMapper.toProfileDto(entity);
+
+        //ASSERT
+
```
Comment (0xD34DC0DE) [inline backend/src/test/java/com/team4/backend/mapping/MonitorMapperTest.java:62]: extra line

File: backend/src/test/java/com/team4/backend/mapping/SupervisorMapperTest.java [hunk 11]
```diff
@@ -56,4 +57,24 @@ void mapEntityToDto() {
...
+    @Test
+    void mapEntityToProfileDto() {
+        //ARANGE
+        Supervisor entity = SupervisorMockData.getMockSupervisor();
+
+        //ACT
+        SupervisorProfileDto dto = SupervisorMapper.toProfileDto(entity);
+
+        //ASSERT
+
```
Comment (0xD34DC0DE) [inline backend/src/test/java/com/team4/backend/mapping/SupervisorMapperTest.java:70]: extra line

**Review by 0xD34DC0DE — COMMENTED** [review]
You decide whether you change them or keep them

**Review by 0xD34DC0DE — APPROVED** [review]

**Pull request merged**